lib/src/core/util/debouncer.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:async';
+
+class Debouncer {
+  Debouncer({required this.milliseconds});
+
+  final int milliseconds;
+  Timer? _timer;
+
+  void run(Function() action) {
+    if (_timer?.isActive ?? false) {
+      _timer?.cancel();
+    }
+    _timer = Timer(Duration(milliseconds: milliseconds), action);
```

Comment:
trailing comma

Reply:
What is the purpose of this Debouncer?

Reply:
The callback used to fetch more movies on scrolling when near the bottom of the list can be called several times a second, but we don't want to call the API every single time because it will get laggy. The Debouncer limits the actual API calls to one per second so that even if the callback is called 20 times in a second, the API is only called once.